src/NCPopupMenu.cc

```diff
@@ -141,3 +171,79 @@ bool NCPopupMenu::postAgain()
     return again;
 }
 
+
+NCursesEvent NCPopupMenu::select_next_item()
+{
+    NCursesEvent event;
+
+    YTableItem * tableItem = dynamic_cast<YTableItem *> ( getCurrentItemPointer() );
+
+    if ( !tableItem)
+	return event;
+
+    auto current = find_item(tableItem);
+
+    if ( current == _items.end() )
+	return event;
+
+    auto begin = std::next(current, 1);
+
+    auto next_enabled = find_if(begin, _items.end(), [](const Item * item) {
+	return item->menu_item->isEnabled() && !item->menu_item->isSeparator();
+    });
+
+    if (next_enabled == _items.end())
+	return event;
+
+    int steps_to_next = std::distance(current, next_enabled);
+
+    for (int i = 0; i < steps_to_next; i++)
+	event = NCPopup::wHandleInput( KEY_DOWN );
+
+    return event;
+}
+
+
+NCursesEvent NCPopupMenu::select_previous_item()
+{
+    NCursesEvent event;
+
+    YTableItem * tableItem = dynamic_cast<YTableItem *> ( getCurrentItemPointer() );
+
+    if ( !tableItem )
+	return event;
+
+    auto current = find_item(tableItem);
+
+    if ( current == _items.end() || current == _items.begin() )
+	return event;
+
+    std::reverse_iterator<std::vector<Item *>::iterator> rbegin(current);
+
+    auto previous_enabled = find_if(rbegin, _items.rend(), [](const Item * item) {
+	return item->menu_item->isEnabled() && !item->menu_item->isSeparator();
+    });
+
+    if (previous_enabled == _items.rend())
+	return event;
+
+    auto previous = find_item((*previous_enabled)->table_item);
+
+    int steps_to_previous = std::distance(previous, current);
+
+    for (int i = 0; i < steps_to_previous; i++)
+	event = NCPopup::wHandleInput( KEY_UP );
+
+    return event;
+}
+
+
+std::vector<NCPopupMenu::Item *>::iterator
+NCPopupMenu::find_item(YTableItem* table_item)
+{
+    auto it = find_if(_items.begin(), _items.end(), [table_item](const Item * item) {
```

Comment:
`auto` plus a hard-to-read lambda plus opening brace not on a separate line as it should be per our C++ coding conventions...